onnxruntime/core/providers/dml/DmlExecutionProvider/src/TensorDesc.cpp

```diff
@@ -315,3 +315,52 @@
     }
     m_bufferTensorDesc.DimensionCount = newDimensionCount;
 }
+
+// Uses dimensionMapping to reorder m_sizes and m_strides to match specific Tensor layout
+void TensorDesc::PermuteDimensions(const std::vector<uint32_t> dimensionMapping, const TensorAxis alignment)
+{
+    EnsureMinimumDimensionCount(dimensionMapping.size(), alignment);
+    InitializeStrides(static_cast<int>(dimensionMapping.size()), alignment);
```

Comment:
>InitializeStrides [](http://example.com/codeflow?start=4&length=17)

We don't want to *overwrite* existing strides, just to *ensure* they exist before permuting, and that must be done before changing the count.

Reply:
Whats the best way to know if strides exists? Any non-zero value in `m_strides` would imply its been populated earlier?

Reply:
Check `m_bufferTensorDesc.Strides` for non-nullptr.